File: saxpy.py
import numpy as np
N = 2 ** 26
XVAL = np.float32(10 * np.random.rand())
YVAL = np.float32(10 * np.random.rand())
AVAL = np.float32(10 * np.random.rand())

def verify(y):
    err = np.sum(np.abs(np.array(y) - (AVAL * XVAL + YVAL)))
    print("Error: {}".format(err))
    return err == 0.0

File: saxpy.m
N = 2 ^ 26
XVAL = 2.5;
YVAL = 1.3;
AVAL = 3.7;

x = ones(N,1) * XVAL;
y = ones(N,1) * YVAL;

tic;
y += x * AVAL;
elapsed = toc;
elapsed = elapsed * 1000

answer = YVAL + AVAL * XVAL;
error = sum(abs(y - answer))
